File: sdk/villow-0.1.0/villow-0.1.0/examples/hello_world_agent.py
from __future__ import annotations

from villow import Agent, Artifact, Composition, Question, preview, task_template


class HelloWorldAgent(Agent):
    @task_template("hello_world")
    def prepare(self, inputs, ctx):
        composition = Composition.input(
            [
                Composition.dropdown_with_descriptions(
                    "tone",
                    label="Tone",
                    options=[{"id": "friendly", "label": "Friendly"}, {"id": "formal", "label": "Formal"}],
                    default="friendly",
                )
            ],
            task_template="hello_world",
            composition_id="hello_world.v1",
        )
        if not ctx.has_answer("tone"):
            return ctx.request_clarification(
                [
                    Question.single_select(
                        id="tone",
                        label="Tone",
                        options=["friendly", "formal"],
                        default="friendly",
                        decide_for_me={"value": "friendly"},
                    )
                ],
                composition_response=composition,
            )
        normalized = {"name": inputs.get("name", "there"), "tone": ctx.answer("tone")}
        return ctx.ready_to_authorize(normalized_inputs=normalized, composition_response=composition)

    @preview("hello_world")
    def preview_hello(self, inputs, ctx):
        return {"type": "text", "value": f"Hello {inputs.get('name', 'there')}"}

    @task_template("hello_world")
    async def run(self, inputs, ctx):
        await ctx.stage_artifact(Artifact.generic(payload={"message": f"Hello {inputs['name']}", "tone": inputs["tone"]}))

File: sdk/villow-sdk-0.1.0/package/examples/hello-world-agent.ts
import { Agent, Artifact, Composition, Question, preview, taskTemplate } from '../src/index.js';

export class HelloWorldAgent extends Agent {
  @taskTemplate('hello_world')
  prepare(inputs: Record<string, unknown>, ctx: { hasAnswer: (id: string) => boolean; answer: (id: string) => unknown; requestClarification: Function; readyToAuthorize: Function }) {
    const composition = Composition.input([
      Composition.dropdownWithDescriptions('tone', {
        label: 'Tone',
        options: [{ id: 'friendly', label: 'Friendly' }, { id: 'formal', label: 'Formal' }],
        default: 'friendly',
      }),
    ], { taskTemplate: 'hello_world', compositionId: 'hello_world.v1' });
    if (!ctx.hasAnswer('tone')) {
      return ctx.requestClarification([
        Question.singleSelect({
          id: 'tone',
          label: 'Tone',
          options: ['friendly', 'formal'],
          default: 'friendly',
          decideForMe: { value: 'friendly' },
        }),
      ], { compositionResponse: composition });
    }
    return ctx.readyToAuthorize({ normalizedInputs: { name: inputs.name ?? 'there', tone: ctx.answer('tone') }, compositionResponse: composition });
  }

  @preview('hello_world')
  previewHello(inputs: Record<string, unknown>) {
    return { type: 'text', value: `Hello ${inputs.name ?? 'there'}` };
  }

  @taskTemplate('hello_world')
  async run(inputs: Record<string, unknown>, ctx: { stageArtifact: Function }) {
    await ctx.stageArtifact(Artifact.generic({ payload: { message: `Hello ${inputs.name}`, tone: inputs.tone } }));
  }
}
